src/ec2/explorer/ec2InstanceNode.ts

```diff
@@ -24,7 +25,7 @@ export class Ec2InstanceNode extends AWSTreeNodeBase implements AWSResourceNode
         public readonly client: Ec2Client,
         public override readonly regionCode: string,
         private readonly partitionId: string,
-        protected instance: Ec2Instance
+        public instance: Ec2Instance
```

Comment:
Is there a specific reason we want this one to be `readonly` or is it generally best practice to make anything `readonly` that isn't changed after initialization?

Reply:
Just the latter

Reply:
In this case the `Ec2Instance` field associated with an `Ec2InstanceNode` is changing because of status updates. The `Ec2Instance` object will have its `status` field updated as the node starts, stops, or restarts. I could refactor to separate the two, but that could get confusing as the original assignment would hold the original status with no indication that it is not being updated or is outdated.

Reply:
I refactored it so that it is read-only, but left a comment that the status attribute is being updated.